virttest/utils_disk.py

```diff
@@ -182,6 +190,559 @@ def clean_old_image(image):
+    for i in range(n_partitions):
+        if i == 0:
+            create_partition_windows(session, did, 5, str(partition_size) + size[-1])
+        else:
+            if part_type == PARTITION_TYPE_EXTENDED:
+                create_partition_windows(session, did, 0, str(extended_size) + size[-1], part_type)
+                part_type = PARTITION_TYPE_LOGICAL
+                create_partition_windows(session, did, 0, str(partition_size) + size[-1], part_type)
+            else:
+                create_partition_windows(session, did, 0, str(partition_size) + size[-1], part_type)
+        drive_letter = set_drive_letter(session, did, partition_no=i + 1)
+        if not drive_letter:
+            return []
+        mountpoint.append(drive_letter)
+        create_filesystem_windows(session, mountpoint[i], fstype)
+    return mountpoint
+
+
+def format_linux_disk(session, did, size, n_partitions=1, fstype="ext4",
+                      labeltype=PARTITION_TABLE_TYPE_MBR):
+    """
+    Create partition on disk in linux guest, format and mount it.
+    Only handle an empty disk and will create equal size partitions onto the disk.
+
+    :param session: session object to guest.
+    :param did: disk kname. e.g. sdb
+    :param size: partition size. e.g. 2G
+    :param n_partitions: the number of partitions on disk
+    :param fstype: filesystem type for the disk.
+    :param labeltype: label type for the disk.
+    :return a list: mount point list for all partitions.
+    """
+    mountpoint = []
+    create_partition_table_linux(session, did, labeltype)
+    partition_size = float(size[:-1]) / n_partitions
+    extended_size = float(size[:-1]) - partition_size
+    start = 0
+    end = partition_size
+    if labeltype == PARTITION_TABLE_TYPE_MBR and n_partitions > 1:
+        part_type = PARTITION_TYPE_EXTENDED
+    else:
+        part_type = PARTITION_TYPE_PRIMARY
+    for i in range(1, n_partitions+1):
+        if i == 1:
+            create_partition_linux(session, did, str(start) + size[-1],
+                                   str(end) + size[-1])
+        else:
+            if part_type == PARTITION_TYPE_EXTENDED:
+                create_partition_linux(session, did, str(start) + size[-1],
+                                       str(extended_size) + size[-1], part_type)
+                part_type = PARTITION_TYPE_LOGICAL
+                create_partition_linux(session, did, str(start) + size[-1],
+                                       str(end) + size[-1], part_type)
+            else:
+                create_partition_linux(session, did, str(start) + size[-1],
+                                       str(end) + size[-1], part_type)
+        start, end = end, end + partition_size
+        create_filesyetem_linux(session, did + str(i), fstype)
+        mount_dst = "/mnt/" + did + str(i)
+        session.cmd("rm -rf %s; mkdir %s" % (mount_dst, mount_dst))
+        mount("/dev/%s" % did + str(i), mount_dst, fstype=fstype, session=session)
```

Comment:
Should check if the status of `mount()`.

Reply:
updated